src/python/controller/extract/extract.py

```
@@ -76,16 +74,34 @@ def verify_archive(archive_path: str):
+        """
+        try:
+            with open(archive_path, 'rb') as f:
+                header = f.read(8)
+            for signature, name in _ARCHIVE_SIGNATURES:
+                if header[:len(signature)] == signature:
+                    return name
+        except OSError:
+            return None
+        # Also check if it's a plain tar (no magic bytes in _ARCHIVE_SIGNATURES)
+        try:
+            with tarfile.open(archive_path) as tf:
```

Ruff (0.15.4)

[error] Local variable `tf` is assigned to but never used

Remove assignment to unused variable `tf`

(F841)